src/snowcli/cli/common/snow_cli_global_context.py

```diff
@@ -88,8 +88,17 @@ def update_global_context(
     def get_connection(self):
         return self.get_global_context_copy().connection.build_connection()
 
+    def execute_string(self, *args, **kwargs):
+        return self.get_connection().ctx.execute_string(*args, **kwargs)
 
-def _create_snow_cli_global_context_manager_with_default_values() -> SnowCliGlobalContextManager:
+    def get_single_value(self, query: str):
+        cursor = self.execute_string(query)
+        return cursor.fetchone()[0]
```

Comment:
I think the direction is fine. But I would suggest to add those methods to `ConnectionDetails` class (and probably rename it?) so we have better separation of responsibilities. The usage would be:
```py
snow_cli_global_context_manager.connection.execute_string(...)
```

Reply:
Thinking more about this, I think these are both the wrong solutions, because right now a new connection will be established with every `execute_string` call.

Does it make sense to have a global connection singleton? How are we preventing multiple connections from being made over the entire lifecycle of a SnowCLI invocation right now?

Reply:
What about wrapping this function or some other in a `@cache` or `@cache_property` [decorator](https://docs.python.org/3/library/functools.html#functools.cache)?

Reply:
Definitely could do. The global context config never changes once it's set up then, I assume?

Reply:
I think so, but I'm not 100% sure.  CC @sfc-gh-mkeller

Reply:
I saw an `update_global_context` function, so I made it clear the cached connection if that's ever called. Felt clearer to implement by hand rather than with `@cache_property` / `del self.get_connection`, lmk what you think!

Reply:
Ah, so if I'm reading that correctly, you just cache it in an instance variable?  If so SGTM.